pkg/registry/core/service/ipallocator/allocator_test.go

```diff
@@ -434,10 +434,18 @@ func TestClusterIPMetrics(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error creating CidrSet: %v", err)
 	}
+	a.EnableMetrics()
 	// create IPv6 allocator
 	cidrIPv6 := "2001:db8::/112"
 	_, clusterCIDRv6, _ := netutils.ParseCIDRSloppy(cidrIPv6)
 	b, err := NewInMemory(clusterCIDRv6)
+	b.EnableMetrics()
+	if err != nil {
+		t.Fatalf("unexpected error creating CidrSet: %v", err)
+	}
+	// create metrics disabled IPv4 allocator
+	// this metrics should be ignored
+	c, err := NewInMemory(clusterCIDRv4)
```

Comment:
can we move this to an individual test instead of overloading current test?
create a with metrics
create b without metrics
check b doesn't update a's metrics

Reply:
I moved this to an individual test.

https://github.com/kubernetes/kubernetes/compare/f2f925a67f91f496e3875b8a8131f7eb243e352e..187af7781a4d6a308ea7f73dd22d435575873396#diff-3ab73f6c7915a3abe7f79d4b8fca55b7c8ee114cbefea86e5b6b849d6c649b12R601